benchexec/tools/ultimate.py

```diff
@@ -105,24 +110,24 @@ def _ultimate_version(self, executable):
                 "--version",
             ],
             # 1
-            [
-                self.get_java(),
-                "-Xss4m",
-                "-jar",
-                launcher_jar,
-                "-data",
-                data_dir,
-                "--version",
-            ],
+            ["-Xss4m", "-jar", launcher_jar, "-data", data_dir, "--version"],
         ]
 
         self.api = len(cmds)
         for cmd in cmds:
-            version = self._query_ultimate_version(cmd, self.api)
-            if version != "":
-                return version
+            for java_version, java in java_versions.items():
```

Comment:
`flake8` just told me that `java_version` is not used here. Is this intended or did you intend to check only specific combinations of commands and Java versions?

If the former:
```suggestion
            for java in java_versions.values():
```

Reply:
Its more for debugging purposes. I need to try all the versions because I cannot determine which one will work without running it.